test/nodes/test_circuit_breaker.py

```
@@ -0,0 +1,719 @@
+            # Don't actually sleep to keep tests fast.
+
+        call_count = 0
+
+        @retry_with_backoff(max_retries=3, base_delay=0.5, max_delay=100.0)
+        def fail():
+            nonlocal call_count
+            call_count += 1
+            raise RateLimitError()
+
+        with patch('llm_base.resilience.time.sleep', side_effect=recording_sleep):
+            with pytest.raises(RateLimitError):
```

Ruff (0.15.7)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)